### ISSUE TITLE
Object may not exist when deleting aci_action_rule_additional_communities

### ISSUE THREAD

**Issue by cpaggen** [CiscoDevNet/terraform-provider-aci#871]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

Terraform v1.0.10
on linux_amd64
+ provider registry.terraform.io/ciscodevnet/aci v2.2.0

Your version of Terraform is out of date! The latest version
is 1.1.9. You can update by downloading from https://www.terraform.io/downloads.html

### APIC version and APIC Platform

4.2(7)

### Affected Resource(s)

aci_action_rule_additional_communities

### Terraform Configuration Files
```hcl
resource "aci_action_rule_additional_communities" "CREATE_SET-SNJ-AWS-NONPROD-SECONDARY-ADDIONAL-COMMUNITY_3" {
   action_rule_profile_dn  = aci_action_rule_profile.CREATE_SET-SNJ-AWS-NONPROD-SECONDARY.id
   community  = "regular:as2-nn2:65100:601"
   annotation = "orchestrator:terraform"
   set_criteria = "append"
   description = "SUMMARY_COMMUNITY"
 }
```

### Debug Output

27140 2022-05-13T16:12:43.207+0200 [INFO]  provider.terraform-provider-aci_v2.2.0: 2022/05/13 16:12:43 [DEBUG] HTTP response unique string GET https://10.48.168.3/api/node/mo/uni/tn-common/attr-SET-SNJ-AWS-NON      PROD-SECONDARY-MS/saddcomm-regular:as2-nn2:65100:601.json {"totalCount":"0","imdata":[]}: timestamp=2022-05-13T16:12:43.207+0200
27141 2022-05-13T16:12:43.207+0200 [INFO]  provider.terraform-provider-aci_v2.2.0: 2022/05/13 16:12:43 [DEBUG] Exit from Do method: timestamp=2022-05-13T16:12:43.207+0200
27142 2022-05-13T16:12:43.207+0200 [INFO]  provider.terraform-provider-aci_v2.2.0: 2022/05/13 16:12:43 [DEBUG] Exit from GET /api/node/mo/uni/tn-common/attr-SET-SNJ-AWS-NONPROD-SECONDARY-MS/saddcomm-regular:as      2-nn2:65100:601.json: timestamp=2022-05-13T16:12:43.207+0200
27143 2022-05-13T16:12:43.207+0200 [INFO]  provider.terraform-provider-aci_v2.2.0: 2022/05/13 16:12:43 [DEBUG] Exit from error (Empty response) {"imdata":[],"totalCount":"0"}: timestamp=2022-05-13T16:12:43.207      +0200
27144 2022-05-13T16:12:43.207+0200 [INFO]  provider.terraform-provider-aci_v2.2.0: 2022/05/13 16:12:43 [DEBUG] uni/tn-common/attr-SET-SNJ-AWS-NONPROD-SECONDARY-MS/saddcomm-regular:as2-nn2:65100:103: Read finis      hed successfully: timestamp=2022-05-13T16:12:43.207+0200
27145 2022-05-13T16:12:43.207+0200 [TRACE] vertex "aci_action_rule_additional_communities.CREATE_SET-SNJ-AWS-NONPROD-SECONDARY-ADDIONAL-COMMUNITY_3": visit complete
27146 2022-05-13T16:12:43.207+0200 [TRACE] NodeAbstractResouceInstance.writeResourceInstanceState to refreshState for aci_action_rule_additional_communities.CREATE_SET-SNJ-AWS-NONPROD-SECONDARY-ADDIONAL-COMMUN      ITY_1
27147 2022-05-13T16:12:43.207+0200 [TRACE] NodeAbstractResouceInstance.writeResourceInstanceState: writing state object for aci_action_rule_additional_communities.CREATE_SET-SNJ-AWS-NONPROD-SECONDARY-ADDIONAL-      COMMUNITY_1
27148 2022-05-13T16:12:43.207+0200 [TRACE] vertex "aci_action_rule_additional_communities.CREATE_SET-SNJ-AWS-NONPROD-SECONDARY-ADDIONAL-COMMUNITY_3": dynamic subgraph encountered errors: Error retrieving Objec      t: Object may not exists
27149 2022-05-13T16:12:43.207+0200 [TRACE] vertex "aci_action_rule_additional_communities.CREATE_SET-SNJ-AWS-NONPROD-SECONDARY-ADDIONAL-COMMUNITY_3": visit complete
27150 2022-05-13T16:12:43.207+0200 [TRACE] vertex "aci_action_rule_additional_communities.CREATE_SET-SNJ-AWS-NONPROD-SECONDARY-ADDIONAL-COMMUNITY_3 (expand)": dynamic subgraph encountered errors: Error retriev      ing Object: Object may not exists
27151 2022-05-13T16:12:43.207+0200 [TRACE] vertex "aci_action_rule_additional_communities.CREATE_SET-SNJ-AWS-NONPROD-SECONDARY-ADDIONAL-COMMUNITY_3 (expand)": visit complete

### Expected Behavior

Terraform creates the resource fine. User manually deletes the community match in the UI then runs terraform plan. Terraform should re-create the resource.

### Actual Behavior

Terraform errors out with "object may not exists" and stops. Resource is not re-created after manual deletion.

### Steps to Reproduce

Create the resource, manually delete in the UI, re-run terraform plan.

**Comment by cpaggen** [comment]
The entire code snippet is at [http://cpaggen-web.cisco.com:9999/cpaggen/capital_one/-/blob/communities/chau.tf] (cisco internal only - customer data)

**Comment by cpaggen** [comment]
The exact same issue happens with resource _aci_match_rule_; as a matter of fact it happens with most recent L3out resources.

**Comment by lhercot** [comment]
I checked and all the case mentioned are fixed in PR #869
